### PR TITLE
some changes in SoftLayer_Virtual_Guest_Template ,needed by bosh-softlayer-cpi 

### PR DESCRIPTION
**Pull request by wen-quan**
According the SL API Doc(http://sldn.softlayer.com/reference/services/SoftLayer_Virtual_Guest/createObject), added some miss properties in SoftLayer_Virtual_Guest_Template.

The modification is needed by the Create method in softlayer_creator.go in bosh-softlayer-cpi project ,otherwise, the bosh-softlayer-cpi 's build & CI will fail.

### CODE REVIEW

File: data_types/softlayer_virtual_guest.go [hunk 1]
```diff
@@ -54,31 +54,29 @@ type SoftLayer_Virtual_Guest_Template_Parameters struct {
 	Parameters []SoftLayer_Virtual_Guest_Template `json:"parameters"`
 }
 
+
+//According the SL API Doc(http://sldn.softlayer.com/reference/services/SoftLayer_Virtual_Guest/createObject),
```
Comment (maximilien) [inline data_types/softlayer_virtual_guest.go]: Cool, you can remove this comment in code. The comment in the PR comment section is good enough

File: data_types/softlayer_virtual_guest.go [hunk 2]
```diff
@@ -54,31 +54,29 @@ type SoftLayer_Virtual_Guest_Template_Parameters struct {
...
 	Datacenter        Datacenter `json:"datacenter"`
 	HourlyBillingFlag bool       `json:"hourlyBillingFlag"`
 	LocalDiskFlag     bool       `json:"localDiskFlag"`
-
-	//Conditionally required
-	OperatingSystemReferenceCode string                    `json:"operatingSystemReferenceCode,omitempty"`
-	BlockDeviceTemplateGroup     *BlockDeviceTemplateGroup `json:"blockDeviceTemplateGroup,omitempty"`
-
-	//Optional
 	DedicatedAccountHostOnlyFlag   bool                            `json:"dedicatedAccountHostOnlyFlag,omitempty"`
```
Comment (maximilien) [inline data_types/softlayer_virtual_guest.go]: Why did you remove the Optional comment. These are the type of comments that are good in code. Simple: `//Required`, `//Optional`, etc and maybe a simple link to documentation.

**Comment by maximilien** [comment]
Hi, @wen-quan,

I have a couple minor comment. Can you please address?

Also, I am wondering if we can have a test that these new properties are loaded. For instance adding missing properties to this [`SoftLayer_Virtual_Guest_Service_createObject.json` template](https://github.com/maximilien/softlayer-go/blob/master/test_fixtures/services/SoftLayer_Virtual_Guest_Service_createObject.json) and modifying this [`#CreateObject` test](https://github.com/maximilien/softlayer-go/blob/master/services/softlayer_virtual_guest_test.go) so that we check for the new properties as well.

Once you do, please rebase, resubmit your PR and if all is well with Travis, I will merge.

Thanks much. Best,

@maximilien

**Comment by wen-quan** [comment]
Dear Max,
Great thanks for your reminder. 
I will follow your suggestion. Thanks!

**Comment by maximilien** [comment]
@wen-quan,

Travis is still failing. See the error:

```
Unit Testing packages:
# github.com/maximilien/softlayer-go/services_test
services/softlayer_virtual_guest_test.go:80: cannot use data_types.PrimaryNetworkComponent literal (type *data_types.PrimaryNetworkComponent) as type data_types.PrimaryNetworkComponent in field value
services/softlayer_virtual_guest_test.go:84: cannot use data_types.PrimaryBackendNetworkComponent literal (type *data_types.PrimaryBackendNetworkComponent) as type data_types.PrimaryBackendNetworkComponent in field value
ok  	github.com/maximilien/softlayer-go/client	0.021s
?   	github.com/maximilien/softlayer-go/client/fakes	[no test files]
?   	github.com/maximilien/softlayer-go/data_types	[no test files]
?   	github.com/maximilien/softlayer-go/main	[no test files]
FAIL	github.com/maximilien/softlayer-go/services [build failed]
?   	github.com/maximilien/softlayer-go/softlayer	[no test files]
SUITE FAILURE
```

**Comment by maximilien** [comment]
I am also closing this. Please address and update and resubmit.
